src/pubkey.cpp

```diff
@@ -334,6 +335,21 @@ bool CPubKey::Derive(CPubKey& pubkeyChild, ChainCode &ccChild, unsigned int nChi
     return true;
 }
 
+std::optional<EllSwiftPubKey> CPubKey::EllSwiftEncode(const std::array<uint8_t, 32>& rnd32) const
+{
+    assert(secp256k1_context_verify && "secp256k1_context_verify must be initialized to use CPubKey.");
+
+    secp256k1_pubkey pubkey;
+
+    if (!secp256k1_ec_pubkey_parse(secp256k1_context_verify, &pubkey, vch, size())) {
+        return {};
+    }
+
+    EllSwiftPubKey encoded_pubkey;
+    secp256k1_ellswift_encode(secp256k1_context_verify, encoded_pubkey.data(), &pubkey, rnd32.data());
```

Comment:
I left this also as a comment on a different PR, perhaps the wrong one.

There shouldn't be a need to invoke `secp256k1_ellswift_encode` here, or have this as a method on `CPubKey`. The new `secp256k1_ellswift_create` function can go directly from private key to ellswift encoding, without needing to convert to a normal public key in the mean time. I think this would be preferable to have as a function on `CKey` instead, and not involve any `CPubKey` at all.

`secp256k1_ellswift_create` is safer (doesn't need additional randomness) and there are some (currently unimplemented) optimizations that may make it faster than the combination of `secp256k1_ec_pubkey_create` + `secp256k1_ellswift_encode`.

Reply:
Done! Thanks for the reminder.